### PR TITLE
Update SubmitCGAP to accommodate new CGAP permissions (C4-371)

### PR DESCRIPTION
**Pull request by netsettler**
This PR is to address [C4-371 ](https://hms-dbmi.atlassian.net/browse/C4-371)...

* Default the institution correctly for the new way a user entity is structured since the permissions change.
* Bring tests up to date.
* Update library dependencies.
* Add a `CHANGELOG.rst`.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -13,11 +13,29 @@ packages = [
...
   'License :: OSI Approved :: MIT License',
+
   'Natural Language :: English',
+
+  # Specify the Python versions you support here. In particular, ensure
+  # that you indicate whether you support Python 2, Python 3 or both.
+  'Programming Language :: Python :: 3',
   'Programming Language :: Python :: 3.6',
+
+  'Framework :: Pyramid',
```
Comment (willronchetti) [inline pyproject.toml:38]: Not exactly accurate WRT this repo but not sure it matters.

File: submit_cgap/submission.py [hunk 2]
```diff
@@ -108,39 +108,48 @@ def get_defaulted_institution(institution, user_record):
...
-            raise SyntaxError("Your user profile declares no institution"
-                              " on behalf of which you are authorized to make submissions.")
-        elif len(submits_for) > 1:
-            raise SyntaxError("You must use --institution to specify which institution you are submitting for"
-                              " (probably one of: %s)." % ", ".join([x['@id'] for x in submits_for]))
-        else:
-            institution = submits_for[0]['@id']
-            show("Using institution:", institution)
+        # Ref: https://hms-dbmi.atlassian.net/browse/C4-371
+        raise SyntaxError("There not currently a way to infer an institution for user,"
```
Comment (willronchetti) [inline submit_cgap/submission.py:121]: `There is not...` ? and I think there is now, it's `user_institution` - but Sarah can comment more specifically on this I think

File: submit_cgap/submission.py [hunk 3]
```diff
@@ -108,39 +108,48 @@ def get_defaulted_institution(institution, user_record):
...
 
 def get_defaulted_project(project, user_record):
     """
     Returns the given project or else if none is specified, it tries to infer a project.
 
-    :param project: the @id of a project
+    :param project: the @id of an project, or None
     :param user_record: the user record for the authorized user
-    :return: the @id of a project to use
+    :return: the @id of an project to use
```
Comment (willronchetti) [inline submit_cgap/submission.py:132]: Why the change from `a` to `an`? Isn't `a` correct?

**Review by willronchetti — APPROVED** [review]
Nothing too crazy in this change - approving with small comments

File: submit_cgap/submission.py (hunk 2 above)
Comment (sbreiff) [reply submit_cgap/submission.py:121]: Yes, I created this field after I created the ticket and didn't remember to update the ticket. Ideally we would like to use user_institution, but can use --institution in the meantime if that change would take longer to make and merge.

Comment (netsettler) [reply submit_cgap/submission.py:121]: I think it won't take long to fix this.

Comment (netsettler) [reply submit_cgap/submission.py]: This should be set.

File: submit_cgap/submission.py (hunk 3 above)
Comment (netsettler) [reply submit_cgap/submission.py:131]: It's far weirder than that. It's that the definitions of the tester for institution and project almost swapped places, so I borrowed the tester for institution with name substitution, and I forgot to change an article.

File: pyproject.toml (hunk 1 above)
Comment (netsettler) [reply pyproject.toml:38]: I agree. Doesn't need to be there.

I added Python 3.7 at the same time, since that's the number we define farther down.

File: submit_cgap/submission.py (hunk 3 above)
Comment (netsettler) [reply submit_cgap/submission.py:131]: Uploading a fix.
